feat: Create initial kubernetes-replicator package

philroche opened the pull request:
This includes remediating all CVEs found after initial build.

Smoke test added to ensure build was successful.

Signed-off-by: philroche <phil.roche@chainguard.dev>

philroche committed:
feat: Create initial kubernetes-replicator package

This includes remediating all CVEs found after initial build.

Smoke test added to ensure build was successful.

Signed-off-by: philroche <phil.roche@chainguard.dev>

kranurag7 commented on kubernetes-replicator.yaml:
```diff
@@ -0,0 +1,38 @@
+    with:
+      packages: .
+      ldflags: |
+        -s -w
```
```suggestion

```
`-s` is something we avoid and `-w` comes out of the box from `go/build` pipeline. ref: https://github.com/chainguard-dev/melange/blob/0900229dc8a4b90e8ad950ffb44e0acee502b996/pkg/build/pipelines/go/build.yaml#L58-L62

kranurag7 replied:
looks like upstream don't set any version related ldflags in their goreleaser config and I was not able to find the other variable as well so we can omit the other ldflag as well.  

https://github.com/mittwald/kubernetes-replicator/blob/master/.goreleaser.yml

philroche replied:
Thank you. Now resolved.

kranurag7 commented on kubernetes-replicator.yaml:
```diff
@@ -0,0 +1,38 @@
+        -X main.version=${{package.version}}
+      output: kubernetes-replicator
+
+  - uses: strip
```
```suggestion

```
this can be removed as well given we already have `-w` via go/build pipeline.

philroche replied:
Thank you. Now resolved.

philroche committed:
fix(kubernetes-replicator): Remove `uses: strip`

`uses: strip` is already part of the go/build pipeline so is not required as a separate step.

Signed-off-by: philroche <phil.roche@chainguard.dev>

philroche committed:
fix(kubernetes-replicator): Remove `-s -w` ldflags

ldflags `-s -w` are already part of the go/build pipeline so is not required to be set prior to build.

Signed-off-by: philroche <phil.roche@chainguard.dev>

kranurag7 commented on kubernetes-replicator.yaml:
```diff
@@ -0,0 +1,35 @@
+  - uses: go/build
+    with:
+      packages: .
+      ldflags: |
+        -X main.version=${{package.version}}
```
I don't see upstream setting this or exposing this variable to set the variable in their main.go at root of the repo so I think we can omit this. 

But this is good to have not a blocker given `go build` with wrong/any ldflags fails silently.

philroche replied:
All build flags now removed. Thank you

philroche committed:
fix(kubernetes-replicator): Remove all build flags from kubernetes-replicator as not required

upstream is not using any ldflags to expose information like verison, commit, date etc.
We can omit passing ldflags from both fips and non-fips packages.

Signed-off-by: philroche <phil.roche@chainguard.dev>
